### PR TITLE
Added support for StyleCop+

### PR DESCRIPTION
**Pull request by Mailaender**
I disabled most of the rules for now just because we violate it so often especially CheckAllowedIndentationCharacters which checks for spaces instead of tabs, but is my next priority.

### CODE REVIEW

File: OpenRA.Game/FileFormats/ImaAdpcmLoader.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public static class ImaAdpcmLoader
 			1282, 1411, 1552, 1707, 1878, 2066, 2272, 2499, 2749,
 			3024, 3327, 3660, 4026, 4428, 4871, 5358, 5894, 6484,
 			7132, 7845, 8630, 9493, 10442, 11487, 12635, 13899, 15289,
-			16818, 18500, 20350, 22385, 24623, 27086, 29794, 32767 
+			16818, 18500, 20350, 22385, 24623, 27086, 29794, 32767
```
Comment (Mailaender) [inline OpenRA.Game/FileFormats/ImaAdpcmLoader.cs]: I just noticed that this seems to be copy pasta code from `AudLoader.cs`, but I am not going to fix the redundancy here in this pull request.

**Comment by obrakmann** [comment]
What's the difference between StyleCop and StyleCop+? And does the `StyleCopConsole` class that the utility uses to run the checks use both?

Looks good otherwise.

**Comment by Mailaender** [comment]
It just adds additional rules. See http://stylecopplus.codeplex.com/ Dumping the DLL makes it work though the MonoDevelop plugin does not ship it which is why those rules are ignored in the IDE.

**Comment by obrakmann** [comment]
Ok then. :+1:

**Comment by pchote** [comment]
Cool, thanks for adding this.  The code changes look good, but I haven't tested it.

**Comment by Mailaender** [comment]
This is actually just removing whitespaces. No code changes to test.

**Comment by pchote** [comment]
There are the additional rules for `--check-code-style` to apply.

**Comment by huwpascoe** [comment]
:+1:

**Comment by Mailaender** [comment]
Yes, all the auto-generated WinForms designer files are ignored now. They contain a lot of whitespace errors created by Visual Studio.

**Comment by Mailaender** [comment]
:bulb: **ProTip!** Add the StyleCopPlus.dll from the thirdparty folder to ~/.local/share/MonoDevelop-5.0/LocalInstall/Addins/MonoDevelop.StyleCop.1.0.1.7/ for IDE integration:

image

**Comment by Mailaender** [comment]
CheckAllowedIndentationCharacters is currently blocked by https://stylecopplus.codeplex.com/workitem/9348 sadly.

**Comment by Mailaender** [comment]
Filed https://github.com/shuruev/StyleCopPlus/pull/5 to avoid the thousands of false-positives.

**Pull request merged**